src/libs/ErrorUtils.ts

```diff
@@ -0,0 +1,113 @@
+import CONST from '../CONST';
+import DateUtils from './DateUtils';
+import * as Localize from './Localize';
+import Response from '../types/onyx/Response';
+import {ErrorFields} from '../types/onyx/OnyxCommon';
+
+function getAuthenticateErrorMessage(response: Response): string {
+    switch (response.jsonCode) {
+        case CONST.JSON_CODE.UNABLE_TO_RETRY:
+            return 'session.offlineMessageRetry';
+        case 401:
+            return 'passwordForm.error.incorrectLoginOrPassword';
+        case 402:
+            // If too few characters are passed as the password, the WAF will pass it to the API as an empty
+            // string, which results in a 402 error from Auth.
+            if (response.message === '402 Missing partnerUserSecret') {
+                return 'passwordForm.error.incorrectLoginOrPassword';
+            }
+            return 'passwordForm.error.twoFactorAuthenticationEnabled';
+        case 403:
+            if (response.message === 'Invalid code') {
+                return 'passwordForm.error.incorrect2fa';
+            }
+            return 'passwordForm.error.invalidLoginOrPassword';
+        case 404:
+            return 'passwordForm.error.unableToResetPassword';
+        case 405:
+            return 'passwordForm.error.noAccess';
+        case 413:
+            return 'passwordForm.error.accountLocked';
+        default:
+            return 'passwordForm.error.fallback';
+    }
+}
+
+/**
+ * Method used to get an error object with microsecond as the key.
+ * @param error - error key or message to be saved
+ */
+function getMicroSecondOnyxError(error: string): Record<number, string> {
+    return {[DateUtils.getMicroseconds()]: error};
+}
+
+type OnyxDataWithErrors = {
+    errors: Record<string, string>;
```

Comment:
```suggestion
    errors?: Errors;
```

Reply:
To accept onyx structures that has `errors` object and are passed to this function.